mesonpy/_editable.py

```diff
@@ -0,0 +1,70 @@
+import functools
+import importlib.abc
+import os
+import subprocess
+import sys
+
+from types import ModuleType
+from typing import Optional, Union
+
+from mesonpy._compat import Collection, Sequence
+
+
+class MesonpyFinder(importlib.abc.MetaPathFinder):
```

Comment:
Sorry, I'm still a little confused on how this is going to be injected into sys.meta_path. 

Is the plan to write a pth file, that does arbritrary code execution? Or go down the sitecustomize route.

Also, where would this file live? IIUC, for it to be importable by a pth file or sitecustomize, it needs to live in site-packages, or meson-python needs to be installed.

I guess requiring meson-python to be installed is OK???, but if pip doesn't cleanup our pth files/wherever we're injecting this, it could be really bad.

Reply:
I pushed more code, but it still isn't complete. Hopefully it would help clarify things for you.

> Sorry, I'm still a little confused on how this is going to be injected into sys.meta_path.
> 
> Is the plan to write a pth file, that does arbritrary code execution? Or go down the sitecustomize route.
 
`.pth` file that executes code.

> I guess requiring meson-python to be installed is OK???

I think so, given this is for editable installs.

> but if pip doesn't cleanup our pth files/wherever we're injecting this, it could be really bad.

It should, we do not install the `.pth` file ourselves. We create a wheel for the editable install, which pip then installs on the system, `pip uninstall` should clean everything up. Maybe taking a look at https://peps.python.org/pep-0660/ could help clarify how this is supposed to work.